src/test/java/wolox/training/controllers/BookControllerTest.java

```diff
@@ -0,0 +1,181 @@
+package wolox.training.controllers;
+
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.doNothing;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.delete;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.put;
+import static org.springframework.test.web.servlet.result.MockMvcResultHandlers.print;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.jsonPath;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import java.util.Collections;
+import java.util.Optional;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.springframework.http.MediaType;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.setup.MockMvcBuilders;
+import wolox.training.models.Book;
+import wolox.training.repositories.BookRepository;
+
+class BookControllerTest {
+
+    private static final String BOOK_PATH = "/api/books";
+    private static final int BOOK_ID = 0;
+    private static final int INVALID_BOOK_ID = 1;
+    private static final int PAGE_NUMBER = 90;
+
+    private MockMvc mockMvc;
+    private BookRepository bookRepository;
+    private ObjectMapper objectMapper;
+    private Book bookTest;
+
+    @BeforeEach
+    void setUp() {
+
+        // Arrange
+        objectMapper = new ObjectMapper();
+        bookRepository = mock(BookRepository.class);
+
+        BookController bookController = new BookController(bookRepository);
+        mockMvc = MockMvcBuilders.standaloneSetup(bookController).build();
+
+        bookTest = Book.builder().genre("Horror")
+            .author("Juan Osorio")
+            .image("horror.jpg")
+            .title("Title")
+            .subtitle("Subtitle")
+            .publisher("Publisher")
+            .year("2020")
+            .pages(PAGE_NUMBER)
+            .isbn("0909-1234-6789-X")
+            .build();
+
+    }
+
+    @Test
+    void whenFindAll_thenReturnAllBooks() throws Exception {
+
+        // Arrange
+        when(bookRepository.findAll()).thenReturn(Collections.singleton(bookTest));
+
+        // Act - Assert
+        mockMvc.perform(get(BOOK_PATH)
+            .accept(MediaType.APPLICATION_JSON_VALUE)
+            .contentType(MediaType.APPLICATION_JSON_VALUE))
+            .andDo(print()).andExpect(status().isOk())
```

Comment:
You may want to test the size of the collection that you receive, for instance check if it has only one element.